test/appium/tests/medium/test_deeplink_chat_share_profile.py

```diff
@@ -0,0 +1,347 @@
+        current_text = message_input.text
+        if current_text != message_text[:-2]:
+            self.errors.append("Message input text '%s' doesn't match expected '%s'" % (current_text, message_text[:-2]))
+
+        """self.home.just_fyi('Cutting message text from input field')
+        message_input.cut_text()
+        message_input.click()
+        if current_text != message_text[:-4]:
+            self.errors.append("Message input text '%s' doesn't match expected '%s'" % (current_text, message_text[:-4]))"""
+
+        message_input.cut_text()
+
+        self.home.just_fyi('Pasting the cut message back to the input field')
+        message_input.paste_text_from_clipboard()
+        if current_text != message_text[:-2]:
+            self.errors.append("Message input text '%s' doesn't match expected '%s'" % (current_text, message_text[:-2]))
+
+        chat.send_message_button.click()
+
+        chat.chat_element_by_text(message_text[:-2]).wait_for_visibility_of_element(2)
+        chat.get_back_to_home_view()
+
+        self.errors.verify_no_errors()
+
+    @marks.testrail_id(702250)
+    def test_chat_public_clear_history_via_options_and_long_press(self):
+
+        public_long_press = '#public-clear-options'
+        options_chat_name = '#' + self.public_chat_name
+        message = 'test message'
+
+        long_press_chat = self.home.join_public_chat(public_long_press[1:])
+        [long_press_chat.send_message(message) for _ in range(2)]
+        self.home.home_button.double_click()
+
+        self.home.element_by_text(options_chat_name).scroll_to_element()
+        options_chat = self.home.get_chat(options_chat_name)
+        chat_view = options_chat.click()
+        [chat_view.send_message(message) for _ in range(2)]
+        self.home.home_button.double_click()
+
+        self.home.element_by_text(public_long_press).scroll_to_element(direction='up')
+        self.home.just_fyi('Clearing history via long press')
+        self.home.clear_chat_long_press(public_long_press)
+
+        self.home.just_fyi('Clearing history via options')
+        options_chat.click()
+        chat_view.clear_history()
+        if chat_view.element_by_text(message).is_element_displayed():
+            self.errors.append(
+                'Messages in %s chat are still shown after clearing history via options' % options_chat_name)
+
+        self.home.just_fyi("Recheck that history won't reappear after relogin")
+        self.home.relogin()
+        for chat_name in options_chat_name, public_long_press:
+            chat = self.home.get_chat(chat_name).click()
+            if chat.chat_element_by_text(message).is_element_displayed():
+                self.errors.append('Messages in %s chat are shown after clearing history and relaunch' % chat_name)
+            chat.home_button.click()
+
+        self.errors.verify_no_errors()
+
+    @marks.testrail_id(702254)
+    # Should be the last in group as it disables network connection
```

Comment:
can be removed; but also can stay, no harm

Reply:
left this comment for a case if will work on this group and add some extra cases